File: app/src/test/java/com/permissionnanny/dagger/MockAppModule.java
package com.permissionnanny.dagger;

import android.app.Application;
import com.permissionnanny.data.AppPermissionDB;
import com.permissionnanny.data.AppPermissionManager;
import com.permissionnanny.data.Cryo;
import com.permissionnanny.data.OngoingRequestDB;

import static org.mockito.Mockito.mock;

/**
 * {@link AppModule} that returns mocks.
 */
public class MockAppModule extends AppModule {

    public MockAppModule() {
        super(null);
    }

    @Override
    Application provideApplication() {
        return mock(Application.class);
    }

    @Override
    AppPermissionDB providePermissionUsageDatabase(Application app, Cryo cryo) {
        return mock(AppPermissionDB.class);
    }

    @Override
    OngoingRequestDB provideOngoingRequestsDatabase(Application app, Cryo cryo) {
        return mock(OngoingRequestDB.class);
    }

    @Override
    AppPermissionManager provideAppPermissionManager(Application app, AppPermissionDB db, Bus bus) {
        return mock(AppPermissionManager.class);
    }
}

File: app/src/test/kotlin/com/permissionnanny/dagger/MockAppModule.kt
package com.permissionnanny.dagger

import android.app.Application
import com.permissionnanny.App
import com.permissionnanny.data.AppPermissionDB
import com.permissionnanny.data.AppPermissionManager
import com.permissionnanny.data.Cryo
import com.permissionnanny.data.OngoingRequestDB
import org.mockito.Mockito.mock

/**
 * [AppModule] that returns mocks.
 */
class MockAppModule : AppModule(mock(App::class.java)) {

    override fun provideApplication(): Application {
        return mock(Application::class.java)
    }

    override fun providePermissionUsageDatabase(app: Application, cryo: Cryo): AppPermissionDB {
        return mock(AppPermissionDB::class.java)
    }

    override fun provideOngoingRequestsDatabase(app: Application, cryo: Cryo): OngoingRequestDB {
        return mock(OngoingRequestDB::class.java)
    }

    override fun provideAppPermissionManager(app: Application, db: AppPermissionDB, bus: AppModule.Bus): AppPermissionManager {
        return mock(AppPermissionManager::class.java)
    }
}
